asteroid/data/dns_dataset.py

```diff
@@ -33,28 +37,54 @@ def __len__(self):
     def __getitem__(self, idx):
         """Gets a mixture/sources pair.
         Returns:
-            mixture, vstack([source_arrays])
+            (mixture, clean, noise) if self.load_noise is true, else (mixture, clean)
         """
         utt_info = self.mix_infos[self.wav_ids[idx]]
         # Load mixture
-        x = torch.from_numpy(sf.read(utt_info["mix"], dtype="float32")[0])
+        x_np, sr = sf.read(utt_info["mix"], dtype="float32")
+        assert sr == self.sample_rate
+        start, stop = get_wav_random_start_stop(
+            len(x_np), int(self.segment * self.sample_rate) if self.segment is not None else None
+        )
+        x = torch.from_numpy(x_np[start:stop])
```

Comment:
Use `start` and `stop` in soundfile.
You can use this to get the length without loading the full file (just reads the header)
```python
wav = sf.SoundFile(filename)
len(wav)
```

Reply:
Ah, using `utt_info["file_len"]`?

Reply:
IDK, I don't remember if there is the `file_len` field in utt_info.. 
If there is, then let's use it. Otherwise, use `sf.SoundFile` because it doesn't load the file in memory, just read the header and is faster.